components/proxy/conf/Caddyfile

```diff
@@ -22,9 +22,18 @@
 
 	servers {
 		protocols h1 h2 h2c
+		metrics
 	}
 }
 
+# Define a matcher to exclude 502 errors on a specific path
+@exclude502 {
```

Comment:
This'll exclude http requests from caddy metrics if the path and status match (I found while testing in preview that 502's were emitted when workspaces started successfully).

Metrics did not have a label with path, I had to look in logs, and there I saw they were all for supervisor, hence this rule being exclusive to supervisor.

@iQQBot what is the 502 for? I assume it is proxy inspecting supervisor endpoints for workspaces. How come `proxy` is checking the supervisor endpoint for workspaces? I was not expecting this and am just curious.

Reply:
Here are the 502's I am referring:
![image](https://github.com/gitpod-io/gitpod/assets/1272076/0be0d532-16fc-435f-94f5-560c141388b5)

Reply:
The 502 are forwarded by `ws-proxy` as long as the workspace is not up, yet, but we already try to poll the status (until it's there).

But note that to my knowledge Previews are the only setup where we still use the `proxy` -> `ws-proxy` chain (FYI originally we used it in self-hosted to just require a single IP).

Let's go with this for now, and see if we can refine it later.

Reply:
> But note that to my knowledge Previews are the only setup where we still use the proxy -> ws-proxy chain

@geropl Can you share code references to what you're referring? This seems wrong to me, we should try to have parity, if possible.

Reply:
Moving conversation to ENT-268